Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentSchema.cpp

```diff
@@ -104,6 +104,40 @@ namespace AZ::DocumentPropertyEditor
         return Dom::Value(result.GetStringView(), false);
     }
 
+    Dom::Value EnumValueAttributeDefinition::ValueToDom(const EnumConstantValue& attribute) const
+    {
+        Dom::Value result(Dom::Type::Object);
+        result[EntryNameKey] = Dom::Value(attribute.m_name, true);
+        result[EntryValueKey] = Dom::Value(static_cast<uint64_t>(attribute.m_value));
+        return result;
+    }
+
+    AZStd::optional<EnumConstantValue> EnumValueAttributeDefinition::DomToValue(const Dom::Value& value) const
+    {
+        if (!value.IsObject() || !value.HasMember(EntryNameKey) || !value.HasMember(EntryValueKey))
+        {
+            return {};
+        }
+
+        return EnumConstantValue{ value[EntryNameKey].GetString(), static_cast<AZ::u64>(value[EntryValueKey].GetUint64()) };
+    }
+
+    AZ::Dom::Value EnumValueAttributeDefinition::LegacyAttributeToDomValue(void* instance, AZ::Attribute* attribute) const
+    {
+        if (attribute == nullptr)
+        {
+            return AZ::Dom::Value();
+        }
+
+        using EnumConstantBasePtr = AZStd::unique_ptr<SerializeContextEnumInternal::EnumConstantBase>;
+        if (auto attributeData = azdynamic_cast<AttributeData<EnumConstantBasePtr>*>(attribute); attributeData != nullptr)
+        {
+            EnumConstantValue* value = reinterpret_cast<EnumConstantValue*>(attributeData->Get(instance).get());
+            return ValueToDom(*value);
+        }
+        return Dom::Value();
+    }
```

Comment:
To preface the words below, this code properly parses the `EnumValue` attributes, but has to use a `reinterpret_cast` which isn't awesome. I haven't verified the DPE loads properly yet because something is preventing the combo boxes from showing up now (might be serialize vs edit related but I'm unsure) but I'm definitely able to see the values being pulled out in `LegacyAttributeToDomValue` using the debugger.

Overrode this function for `EnumValue`s in order to single them out. Made it easier to debug at first but, once Alex and I realized what was going on, it also made it clear to me that skipping the `AttributeReader` might help show the problem better.

So, the `EnumValue` attribute is stored as a `unique_ptr<SerializeContextEnumInternal::EnumConstantBase>` in its `AttributeData`. I'm assuming this is fine for whatever is done with it internal to the reflection system, but it's causing problems for us trying to pull the data out and into a `Dom::Value`. Problems I was running into:
- Can't copy the `unique_ptr` (as expected for unique_ptr)
- `AttributeData::Get` returns a `const` qualified value; in this case it yields a `const unique_ptr<SerializeContextEnumInternal::EnumConstantBase>` (this might have been causing me problems when trying to `AZStd::move` the `unique_ptr` into a `shared_ptr` so I could try `AZStd::rtti_pointer_cast`
- The dereferenced type is an abstract base class (`EnumConstantBase`) so we can't do much with it other than cast or pass it
- It's a `unique_ptr`, and detecting that an attribute's data is a unique_ptr inside of the `AttributeReader` seems like overkill for this case (I'm assuming not many attributes would be stored this way)

Those problems taken together are making it very hard (seemingly impossible) to use the `AttributeReader` or do anything other than `reinterpret_cast` the base type ptr into something we can use.

I'm sure there is another way to handle this. In fact, I'm still a little curious what all would be involved in turning the context's `EnumValue` attributes into `EnumValues` in the `LegacyReflectionBridge` similar to how we turn all read-only related attributes into `Disabled` in the DPE.